Il2CppInterop.Runtime/Runtime/Il2CppObjectPool.cs

```diff
@@ -40,10 +40,14 @@ public static T Get<T>(IntPtr ptr)
         var newObj = Il2CppObjectBase.InitializerStore<T>.Initializer(ptr);
         unsafe
         {
-            var nativeClassStruct = UnityVersionHandler.Wrap((Il2CppClass*)Il2CppClassPointerStore<T>.NativeClassPtr);
-            if (!nativeClassStruct.HasFinalize)
+            var il2CppClass = (Il2CppClass*)Il2CppClassPointerStore<T>.NativeClassPtr;
+            if (il2CppClass != null)
             {
-                Il2CppSystem.GC.ReRegisterForFinalize(newObj as Object ?? new Object(ptr));
+                var nativeClassStruct = UnityVersionHandler.Wrap(il2CppClass);
+                if (!nativeClassStruct.HasFinalize)
+                {
+                    Il2CppSystem.GC.ReRegisterForFinalize(newObj as Object ?? new Object(ptr));
+                }
             }
```

Comment:
This should be a separate pull request because it's mostly unrelated to your other changes, and it requires a more in-depth review.